src/main/java/mate/academy/dao/impl/AbstractDao.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+
+public abstract class AbstractDao {
+    protected <T> T add(T entity) {
+        HibernateUtil.getSessionFactory().inTransaction(session -> {
```

Comment:
dont use this method. Same in other casses

Reply:
Why not?) It does the same thing under the hood as I would write it usually, but ok, I replaced it.